Update devDependencies, fix new linter errors, and add Node 8 and 10 to test matrix, Remove Node 4 and 5

lencioni opened the pull request:
Just a little bit of regular maintenance.

lencioni committed:
Update devDependencies and fix linter errors

lencioni committed:
Add Node 8 and 10 to test matrix

ljharb commented on __tests__/canUsePassiveEventListeners-test.js:
```diff
@@ -22,74 +22,76 @@ describe('when not canUseDOM', () => {
-      expect(window.addEventListener).toHaveBeenCalledTimes(1);
-    });
+    wrap()
+      .withOverrides(() => window, () => ({
```
༼;´༎ຶ ۝ ༎ຶ༽

ljharb commented on package.json:
```diff
@@ -37,16 +37,16 @@
-    "jest": "^20.0.0",
-    "jest-wrap": "^1.0.0",
-    "rimraf": "^2.6.1",
+    "jest": "^23.1.0",
```
Upgrading jest will probably not be an option since it aggressively drops support for older nodes.

lencioni replied:
Oh interesting. Good to know. I'm okay with not running tests in Node 4 and 5. I'll update the .travis.yml file now. Thanks for the heads up!

ljharb replied:
My suggestion would be to freeze jest, not to drop testing on supported platforms :-(

ljharb replied:
Or, to do like airbnb will be doing for public packages, and switch to tape - since that’s the only test framework that has support for all environments.

ljharb commented:
Note how jest is currently running 0 tests in ci

lencioni committed:
Remove Node 4 and 5 from test matrix

It seems that newer versions of Jest don't support these older versions
of Node. I'd rather keep Jest updated than run tests on these versions,
so here we are.
